plan9/9front.patch

```diff
@@ -0,0 +1,66 @@
+diff --git a/dist/IO/poll.h b/dist/IO/poll.h
+--- a/dist/IO/poll.h
++++ b/dist/IO/poll.h
+@@ -22,10 +22,12 @@
+ 
+ #define EMULATE_POLL_WITH_SELECT
+ 
++#ifdef __VMS
+ #ifdef poll
+ # undef poll
+ #endif
+ #define poll Perl_my_poll
++#endif
+ 
+ typedef struct pollfd {
+     int fd;
+diff --git a/ext/Errno/Errno_pm.PL b/ext/Errno/Errno_pm.PL
+--- a/ext/Errno/Errno_pm.PL
++++ b/ext/Errno/Errno_pm.PL
+@@ -125,6 +125,8 @@ sub get_files {
+     # VMS keeps its include files in system libraries
+     if ($^O eq 'VMS') {
+ 	$file{'Sys$Library:DECC$RTLDEF.TLB'} = 1;
++	} elsif ($^O eq 'plan9') {
++	$file{'/sys/include/ape/errno.h'} = 1;
+     } elsif ($^O eq 'os390') {
+ 	# OS/390 C compiler doesn't generate #file or #line directives
+ 	$file{'/usr/include/errno.h'} = 1;
+diff --git a/sv.c b/sv.c
+--- a/sv.c
++++ b/sv.c
+@@ -876,10 +876,10 @@
+     U8 body_size;	/* Size to allocate  */
+     U8 copy;		/* Size of structure to copy (may be shorter)  */
+     U8 offset;		/* Size of unalloced ghost fields to first alloced field*/
+-    PERL_BITFIELD8 type : 4;        /* We have space for a sanity check. */
```

Comment:
Why avoid the bitfields?

Reply:
Compiler issue -- right now, static initialization of bitfields breaks things. It's something I'm working on slowly, but for now we need to avoid bitfields when variables have initializers.

Reply:
(Also, we could shrink down the patch if we just change 'typestring' to 'typestr' in the main code -- 'typestring' conflicts with a compiler extension)

Reply:
Using bitfields fails to build with the Plan9 C compiler. There may be something else going on, or a bug in the Plan9 C compiler which I haven't yet investigated. So in the interim, this is why I'm having a separate patch for Plan 9 rather than changing sv.c directly--it needs more investigation. @oridb might be able to speak more to it.

Reply:
> Also, we could shrink down the patch if we just change 'typestring' to 'typestr' in the main code -- 'typestring' conflicts with a compiler extension

That sounds like a fine plan.

Reply:
> > Also, we could shrink down the patch if we just change 'typestring' to 'typestr' in the main code -- 'typestring' conflicts with a compiler extension
> 
> That sounds like a reasonable plan, no one is married the name of that variable

Updated with commit 92c4078967.

Reply:
@Leont can we mark this discussion as resolved?

Reply:
@atoomic : note the reason here for changing 'typestring' to 'typestr'.